R/transform-calculate.R

```diff
@@ -0,0 +1,147 @@
+# calculcate transforms for an array.
```

Comment:
Could you please document these functions a bit more verbosely? Would help me to follow along and understand what all this is doing.